packages/hint-compat-api/src/helpers/compat-api.ts

```diff
@@ -0,0 +1,156 @@
+// Waiting for this PR https://github.com/mdn/browser-compat-data/pull/3004
+const bcd: CompatData = require('mdn-browser-compat-data');
+
+import { forEach } from 'lodash';
+import { browserVersions } from './normalize-version';
+import { BrowserSupportCollection, MDNTreeFilteredByBrowsers } from '../types';
+import { CompatData, CompatStatement, SupportStatement, SimpleSupportStatement } from '../types-mdn.temp'; // Temporal
+
+type CompatNamespace = 'css' | 'javascript' | 'html';
+
+export class CompatApi {
+    public compatDataApi: MDNTreeFilteredByBrowsers; // Any because no types by the moment, check line 1
+    private browsers: BrowserSupportCollection;
+
+    public constructor(namespaceName: CompatNamespace, browsers: BrowserSupportCollection, isCheckingNotBroadlySupported = false) {
+        this.browsers = browsers;
+        this.compatDataApi = bcd[namespaceName];
+        this.compatDataApi = this.applyBrowsersConfiguration(isCheckingNotBroadlySupported);
+    }
+
+    private applyBrowsersConfiguration(isCheckingNotBroadlySupported = false): any {
+        const compatDataApi = {} as MDNTreeFilteredByBrowsers;
+
+        forEach(this.compatDataApi, (namespaceFeaturesValues, namespaceFeaturesKey) => {
+            const namespaceFeatures = {} as CompatStatement & MDNTreeFilteredByBrowsers;
+
+            forEach(namespaceFeaturesValues, (featureValue, featureKey) => {
+                const typedFeatureValue = featureValue as CompatStatement & MDNTreeFilteredByBrowsers;
+
+                if (!this.isFeatureRequiredToTest(typedFeatureValue, isCheckingNotBroadlySupported)) {
+                    return;
+                }
+
+                namespaceFeatures[featureKey] = typedFeatureValue;
+            });
+
+            compatDataApi[namespaceFeaturesKey] = namespaceFeatures;
+        });
+
+        return compatDataApi;
+    }
+
+    public getPrefix(name: string): [string | undefined, string] {
+        const regexp = new RegExp(`-(moz|o|webkit|ms)-`, 'gi');
+        const matched = name.match(regexp);
+
+        return matched && matched.length > 0 ? [matched[0], name.replace(matched[0], '')] : [undefined, name];
+    }
+
+    public getSupportStatementFromInfo(browserFeatureSupported?: SupportStatement, prefix?: string): SimpleSupportStatement | undefined {
```

Comment:
I don't think we are memoizing the result of anything (PR is big so I might have missed it) and I imagine we will be checking multiple times the same properties. If that's the case we should look into adding that.

Reply:
Adding as a task